src/components/organisms/Badges/Badges.tsx

```diff
@@ -48,13 +39,36 @@ export const Badges: React.FC<{
           >)
       ) ?? [];
 
+    const venuesRequests = visits.map((visit) =>
+      firestore.collection("venues").where("name", "==", visit.id).get()
+    );
+
+    const venues: WithId<AnyVenue>[] = [];
+    const hasVenuesRequests = isTruthy(venuesRequests);
+
+    // If there are no venues visited avoid sending the request.
+    if (hasVenuesRequests) {
+      const requestSnapshots = await Promise.all(venuesRequests);
```

Comment:
@denisdimitrov This was actually how I was thinking of doing it too. Removes the need to do the 'limited to 10 at a time' batches. The only potential optimisation I considered was that if we use this `promise.all` pattern but also batch with `in` as well. That way we should be able to send 10x less requests (in theory) (though I expect for most people it will be so few requests in reality that it may not even matter that much)

Reply:
@denisdimitrov Is it supposed to be `name == visit.id` ? Seems sketchy, just wanted to confirm

Reply:
@thecadams `visit.id` is actually the venue name / room title.
@0xdevalias Chris already added this with his commit.